crates/sui-source-validation-service/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::{Path, PathBuf};
+
+use actix_web::{dev::Server, web, App, HttpRequest, HttpServer, Responder};
+
+use move_package::BuildConfig as MoveBuildConfig;
+use sui_move::build::resolve_lock_file_path;
+use sui_move_build::{BuildConfig, SuiPackageHooks};
+use sui_sdk::wallet_context::WalletContext;
+use sui_source_validation::{BytecodeSourceVerifier, SourceMode};
+
+pub async fn verify_package(
+    context: &WalletContext,
+    package_path: impl AsRef<Path>,
+) -> anyhow::Result<()> {
+    move_package::package_hooks::register_package_hooks(Box::new(SuiPackageHooks));
+    let config = resolve_lock_file_path(
+        MoveBuildConfig::default(),
+        Some(package_path.as_ref().to_path_buf()),
+    )
+    .unwrap();
+    let build_config = BuildConfig {
+        config,
+        run_bytecode_verifier: false, /* no need to run verifier if code is on-chain */
+        print_diags_to_stderr: false,
+    };
+    let compiled_package = build_config
+        .build(package_path.as_ref().to_path_buf())
+        .unwrap();
+
+    let client = context.get_client().await?;
+    BytecodeSourceVerifier::new(client.read_api())
+        .verify_package(&compiled_package, true, SourceMode::Verify)
```

Comment:
The `true` here means we are verifying dependencies -- do we want to do that?  This flag was originally added because we thought that a source indexer would not need it.

Also, nit: Can we label the boolean to explain what it's for?

```suggestion
        .verify_package(&compiled_package, /* verify_deps */ true, SourceMode::Verify)
```

Reply:
Yeah good question! I think I briefly processed this question when I was coding, and basically thought:

- What are the consequences of us showing "✅ verified" for a package in Sui explorer that _doesn't_ have verified dependencies (i.e., any `use XYZ` outside of this package)
- Could this mean that someone "sneakily" publishes a `sui` package at a different address and creates the perception that a standard `sui` framework function is being called, but instead it does something nefarious?
- That seems really bad, best to verify deps / entire dep chain

But in that brief moment I didn't really consider that the sneakily published `sui` package could also be "verified" regardless, and wouldn't make a difference unless we also reveal the dependencies that a package is verified against. Which is something we might want to think about separately. So granted, after rethinking, in the current scope we can probably live with this being `false`.

Reply:
Sounds good!  Note that the thinking around not verifying deps in the source indexing case was more that the indexer would have verified the dependencies at some other time, so we could avoid doing accidentally quadratic work by re-verifying them here.

When we do support a broader set of packages, we should still care whether a package's dependencies are verified or not, but this matters more from an overall "quality" signal perspective -- it's still fine to display source for a package that is itself verified even if its dependencies are not.